### PR TITLE
Fix larva queue

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

Fix negative burrowed. No bug sighted for now. Ready to merge

## Why It's Good For The Game

Xeno economy suffers from larva debt, stop the larva debt

## Changelog
:cl:
fix: Negative larva fixed
/:cl:

### CODE REVIEW

**Comment by mrhugo13** [comment]
> This might break call of burrowed, as i couldn't test it on local serb

it does

**Comment by BraveMole** [comment]
@TiviPlus ready for merge

File: code/datums/jobs/job/job.dm [hunk 1]
```diff
@@ -312,3 +313,8 @@ GLOBAL_PROTECT(exp_specialmap)
 
 /datum/job/proc/handle_special_preview(client/parent)
 	return FALSE
+
+/datum/job/xenomorph/occupy_job_positions(amount, respawn)
+	if((total_positions - current_positions - amount) < 0)//Bad, really bad
+		return FALSE
+	return ..()
```
Comment (TiviPlus) [inline code/datums/jobs/job/job.dm:320]: is this hack still needed

File: code/modules/mob/living/carbon/xenomorph/hive_datum.dm [hunk 2]
```diff
@@ -590,7 +590,7 @@ to_chat will check for valid clients itself already so no need to double check f
 	var/list/possible_mothers = list()
 	var/list/possible_silos = list()
 	SEND_SIGNAL(src, COMSIG_HIVE_XENO_MOTHER_PRE_CHECK, possible_mothers, possible_silos) //List variable passed by reference, and hopefully populated.
-
+
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/hive_datum.dm:593]:
```suggestion

```

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/datums/jobs/job/job.dm (hunk 1 above)
Comment (BraveMole) [reply code/datums/jobs/job/job.dm:320]: Probably not. I still want to keep this, it's a hard check in case i missed a weird edge case or whatever

Comment (TiviPlus) [reply code/datums/jobs/job/job.dm:320]: You shouldnt let code silently fail like this, remove it

File: code/datums/jobs/job/job.dm [hunk 3]
```diff
@@ -312,3 +313,10 @@ GLOBAL_PROTECT(exp_specialmap)
 
 /datum/job/proc/handle_special_preview(client/parent)
 	return FALSE
+
+/datum/job/xenomorph/occupy_job_positions(amount, respawn)
+	if((total_positions - current_positions - amount) < 0)
+		log_debug("Occupy xenomorph position was call with amount = [amount] and respawn =[respawn ? "TRUE" : "FALSE"] \n \
+		This would have created a negative larva situation")
+		return FALSE
```
Comment (TiviPlus) [inline code/datums/jobs/job/job.dm:321]: Should be a crash or stack trace

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**